### ISSUE TITLE
Muliple NBSPs in code secontion in markdown string are collapsed

### ISSUE THREAD

**Issue by nschonni** [prettier/prettier#13485]
**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAxCAnABHAHgQwFsAHAGzgBosYALOLAZxgwEsoBzLAAwB0R8ABUIBGfLlgAUtDHHoNi+MHAZZhcGAHdZULAH18urPigATPcN0BKRnAUZ88FfsPGzui1kKZ6tY9z74qmKS0HIKSqrqWgh6Bkam5lYAdCAUIBDEMCzQDMig+BgYEBoACgUIuSj4pBr4AJ65acL2YADW6gDKRHAAMmxwyABm1QyUIM2K7TAd4WzsyMwArmNwhGomJnAmPcbsi-jscOgYhA5ZHMj8izAQqSA0MISkAOo0LI7hcB0V7ywAbu86pcwAxGiA2KMMDASvZ2KchiMxgArBi4DpzcgARUWEHgCNIozSdkhl1OGFaJmKUDuxFYsGeLBMtGQAA4AAxEoqjZ72YiXWnKOAYP4DNIARxx8BhGUq-AYAFooLJNiY7jIJSwZDCDvCkMMCWNRoQWAsMMs0gwMXBsbiBnrEWkYPhhAymTRkAAmR32FikOYAYQghF1IGUAFY7otRgAVZ2VfWEkB-ZYASVMCGmYFYmQAgqYOjA6uR8aMAL6loA)

```sh
--parser markdown
```

**Input:**

```markdown
For example, the string `"a   b"` (three spaces between _a_ and _b_) separates _a_ and _b_ more than `"a b"` (one space between _a_ and _b_).
```

**Output:**

```markdown
For example, the string `"a b"` (three spaces between _a_ and _b_) separates _a_ and _b_ more than `"a b"` (one space between _a_ and _b_).

```

**Expected behavior:**
I replaced regular spaces with Non-breaking spaces to prevent the collapsing of the spaces that occurs with regular spaces, but if you look at `"a   b"` it shouldn't be collapsed as it renders correctly.

**Comment by fisker** [comment]
https://github.com/prettier/prettier/blob/cd3e530c2e51fb8296c0fb7738a9afdd3a3a4410/src/language-markdown/print-preprocess.js#L31-L38

**Comment by kachkaev** [comment]
This replace logic dates back to https://github.com/prettier/prettier/pull/2943 ([line](https://github.com/prettier/prettier/pull/2943/files#diff-d7ab6a0f77ca5a7100f25d35f5b893e76ce9ca024f69044a4ae501ede2e8f750R58)). I agree that it’s best not to touch the contents of ``` `` ```, just sharing the origin for historical context.

**Comment by nschonni** [comment]
Yeah, and original the doc had 3 regular spaces `"a   b"` that does collapse done, so Prettier is doing the write thing in that case. This appears to just be an odd edge case, and could always be prettier-ignore'd

**Comment by thorn0** [comment]
> original the doc had 3 regular spaces "a b" that does collapse down

It's collapsed by the browser. In the HTML markup, it's still 3 spaces. I'd vote for removing this `replace` unless there really is a case when it's doing the right thing.

Actually, according to the [CommonMark spec](https://spec.commonmark.org/0.30/#code-spans), the only correct replacement would be to replace line endings with spaces.

**Comment by mrienstra** [comment]
Just ran into this while editing a `.md` file, this regex search:
`^(  )+(?![ \n])` (two spaces between first pair of parentheses)
... became:
`^( )+(?![ \n])` (one space between first pair of parentheses)

Another vote for dropping this behavior.

**Comment by kachkaev** [comment]
You are right @thorn0, [interior spaces should not be collapsed](https://spec.commonmark.org/0.30/#example-337), according to CommonMark.

```
`foo   bar 
baz`
```
↓
```
<p><code>foo   bar  baz</code></p>
```

**Comment by kachkaev** [comment]
WIP PR: https://github.com/prettier/prettier/pull/13590 (needs edge case fixing)

**Comment by thorn0** [comment]
Fixed by #13590 in `next`
